aptos-move/aptos-vm/src/sharded_block_executor/local_executor_shard.rs

```diff
@@ -0,0 +1,249 @@
+// Copyright © Aptos Foundation
+
+use crate::sharded_block_executor::{
+    executor_shard::{CoordinatorClient, CrossShardClient, ExecutorClient},
+    messages::CrossShardMsg,
+    sharded_executor_service::ShardedExecutorService,
+    ExecutorShardCommand,
+};
+use aptos_block_partitioner::sharded_block_partitioner::MAX_ALLOWED_PARTITIONING_ROUNDS;
+use aptos_logger::trace;
+use aptos_state_view::StateView;
+use aptos_types::{
+    block_executor::partitioner::{RoundId, ShardId, SubBlocksForShard},
+    transaction::{analyzed_transaction::AnalyzedTransaction, TransactionOutput},
+};
+use crossbeam_channel::{unbounded, Receiver, Sender};
+use move_core_types::vm_status::VMStatus;
+use std::{
+    sync::{Arc, Mutex},
+    thread,
+};
+
+/// Executor service that runs on local machine and waits for commands from the coordinator and executes
+/// them in parallel.
+pub struct LocalExecutorService<S: StateView + Sync + Send + 'static> {
+    _executor_service: Arc<ShardedExecutorService<S>>,
```

Comment:
dead code?

Reply:
We need to keep the executor service in scope for the time `LocalExecutorService` is used. Let me add a comment here.

Reply:
why is it needed? isn't that moved into the thread in L53?

Reply:
You are right, good point. I meant to keep only the join handle here....